src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -38,15 +38,16 @@ internal static partial class CapiHelper
         public static string UpgradeDSS(int dwProvType, string wszProvider)
         {
             string wszUpgrade = null;
-            SafeProvHandle safeProvHandle = SafeProvHandle.InvalidHandle;
             if (string.Equals(wszProvider, MS_DEF_DSS_DH_PROV, StringComparison.Ordinal))
             {
+                SafeProvHandle safeProvHandle;
                 // If this is the base DSS/DH provider, see if we can use the enhanced provider instead.
-                if (S_OK == AcquireCryptContext(ref safeProvHandle, null, MS_ENH_DSS_DH_PROV, dwProvType,
+                if (S_OK == AcquireCryptContext(out safeProvHandle, null, MS_ENH_DSS_DH_PROV, dwProvType,
                                         (uint)CryptAcquireContextFlags.CRYPT_VERIFYCONTEXT))
                 {
                     wszUpgrade = MS_ENH_DSS_DH_PROV;
                 }
+                safeProvHandle.Dispose();
```

Comment:
I was very tempted to replace some `int Method(out SafeSomethingHandle, ...)` methods with `SafeSomethingHandle Method(..., out int hr)` so I could consume them in a using statement.

I was trying to keep the changes down a bit (though, yeah, there were a lot); but if anyone wants to second flipping them I'll happily do it :).